nla_semparse/nla_semparse/nla_metric.py

```diff
@@ -0,0 +1,111 @@
+from typing import Dict, List, Optional
+from overrides import overrides
+from allennlp.training.metrics.metric import Metric
+
+
+@Metric.register('nla_metric')
+class NlaMetric(Metric):
+    """
+    Metric for evaluating prefix arithmetic sequences against targets, useful for Natural Language Arithmetic
+    parsing. This metric evaluates predicted sequences on three things: 1) whether the predicted metric is a
+    well-formed prefix arithmetic expression, 2) whether the predicted sequence and the target seqquence evaluate
+    to the same value, 3) whether the predicted sequence and the target sequence are identical.
+    """
+    def __init__(self):
+        self._operator_logic = {"+": lambda x, y: x + y,
```

Comment:
When we get to grammar-based decoding, this is going to need to be `plus`, not `+`, so that we can write python functions with these names.

Reply:
Yes, the data generation script needs to be changed as well.